tests/test_packages/test_connections/test_p2p_libp2p/test_public_dht.py

```diff
@@ -85,171 +82,74 @@ def delegate_uris_public_keys(request):
-            multiplexer2 = Multiplexer([connection2])
-            self.log_files.append(connection2.node.log_file)
-            multiplexer2.connect()
-            self.multiplexers.append(multiplexer2)
+    def test_connectivity(self):
+        """Test connectivity."""
+        assert self.all_multiplexer_connections_connected
 
-            sender = connection1.address
-            to = connection2.address
+    def test_communication_direct(self):
+        """Test direct communication through the same entry peer"""
 
+        for mux_pair in self.pairs_with_same_entry_peers:
+            sender, to = (c.address for m in mux_pair for c in m.connections)
             envelope = self.enveloped_default_message(to=to, sender=sender)
-            multiplexer1.put(envelope)
-            delivered_envelope = multiplexer2.get(block=True, timeout=30)
+            mux_pair[0].put(envelope)
+            delivered_envelope = mux_pair[1].get(block=True, timeout=30)
             assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
 
-    @pytest.mark.parametrize("maddrs", [PUBLIC_DHT_MADDRS], indirect=True)
-    def test_communication_indirect(self, maddrs):
-        """Test communication indirect."""
-
-        assert len(maddrs) > 1, "Test requires at least 2 public dht node"
-
-        for maddr1, maddr2 in itertools.permutations(maddrs, 2):
-
-            connection1 = _make_libp2p_connection(relay=False, entry_peers=[maddr1])
-            multiplexer1 = Multiplexer([connection1])
-            self.log_files.append(connection1.node.log_file)
-            multiplexer1.connect()
-            self.multiplexers.append(multiplexer1)
-            sender = connection1.address
+    def test_communication_indirect(self):
+        """Test indirect communication through another entry peer"""
 
-            connection2 = _make_libp2p_connection(relay=False, entry_peers=[maddr2])
-            multiplexer2 = Multiplexer([connection2])
-            self.log_files.append(connection2.node.log_file)
-            multiplexer2.connect()
-            self.multiplexers.append(multiplexer2)
-
-            to = connection2.address
+        for mux_pair in self.pairs_with_different_entry_peers:
+            sender, to = (c.address for m in mux_pair for c in m.connections)
             envelope = self.enveloped_default_message(to=to, sender=sender)
-            multiplexer1.put(envelope)
-            delivered_envelope = multiplexer2.get(block=True, timeout=30)
-
+            mux_pair[0].put(envelope)
+            delivered_envelope = mux_pair[1].get(block=True, timeout=30)
             assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
 
 
 @pytest.mark.integration
 @libp2p_log_on_failure_all
-class TestLibp2pConnectionPublicDHTDelegate(BaseP2PLibp2pTest):
-    """Test that public DHT's delegate service is working properly"""
+class TestLibp2pConnectionPublicDHTDelegate(TestLibp2pConnectionPublicDHTRelay):
```

Comment:
note that we inherit from `TestLibp2pConnectionPublicDHTRelay`. Hence we only need to overwrite the `setup_class` method now, and can reuse the test methods already implemented